src/machines/systemIO/systemIOMachine.ts

```diff
@@ -453,10 +453,14 @@ export const systemIOMachine = setup({
             assign({
               requestedFileName: ({ context, event }) => {
                 assertEvent(event, SystemIOMachineEvents.done_importFileFromURL)
-                // Not the entire path
+                console.log(event.output)
+                // Gotcha: file could have an ending of .kcl...
+                const file = event.output.fileName.endsWith('.kcl')
+                  ? event.output.fileName
+                  : event.output.fileName + '.kcl'
```

Comment:
Not sure to fully understand what's going on here?

Reply:
Let me double check tomorrow, I think there was a scenario that the `fileName` in my actor could have an `.kcl` or not. I need to make sure the final result has `.kcl` in it but the `event.output.fileName` internally may or may not have it.

I can come up with a cleaner solution tomorrow.